src/mono/mono/mini/mini.c

```diff
@@ -4602,3 +4605,22 @@ mini_get_simd_type_info (MonoClass *klass, guint32 *nelems)
 		return MONO_TYPE_VOID;
 	}
 }
+
+guint32 mini_number_of_elements(MonoClass *klass){
```

Comment:
As `mini_get_simd_type_info` is already calculating the number of elements, do we really need this additional method?
It seems like it duplicates most of the logic making it harder to maintain.